### PR TITLE
rgw: Adding 'iam' namespace for Role and User Policy related REST APIs.

### PR DESCRIPTION
**Pull request by pritha-srivastava**
Signed-off-by: Pritha Srivastava <prsrivas@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/rgw/rgw_rest_iam.cc [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+#define dout_subsys ceph_subsys_rgw
+
+void RGWHandler_REST_IAM::rgw_iam_parse_input()
+{
+  if (post_body.size() > 0) {
+    ldout(s->cct, 10) << "Content of POST: " << post_body << dendl;
+
+    if (post_body.find("Action") != string::npos) {
+      boost::char_separator<char> sep("&");
+      boost::tokenizer<boost::char_separator<char>> tokens(post_body, sep);
```
Comment (mattbenjamin) [inline src/rgw/rgw_rest_iam.cc]: iirc boost::tokenizer isn't thought to be very efficient--maybe not a concern for now

Comment (mattbenjamin) [reply src/rgw/rgw_rest_iam.cc]: @adamemerson thoughts?

Comment (pritha-srivastava) [reply src/rgw/rgw_rest_iam.cc]: @mattbenjamin @adamemerson @cbodley : I have the exact same method for parsing sts_input also, with a few differences. I could have made this into a common method that parses output for both. But later when we decide to have STS, IAM served at a different hostname:portname, then these methods will already be separate. Is this ok for now?

**Comment by mattbenjamin** [comment]
fine with me

**Comment by pritha-srivastava** [comment]
The teuthology run is here: http://pulpito.ceph.com/prsrivas-2019-04-12_08:57:26-rgw-wip-rgw-iam-distro-basic-smithi/

**Pull request merged**